grafana/grafana
# Glue: Split correlations editor into 3 steps

**What this PR does / why we need it**:

This PR splits the Correlation editor into 3 steps to improve user experience.

* Some copy was adjusted to make it more clear. Happy to get feedback on it though we can revise docs later as this task is blocking other improvements.
* Wizard component is not meant to be used in other places yet. If it proves to work fine we can improve the API (for example adding default navigation and making the layout more flexible) and then think about promoting it to grafana/ui though it's not the main concern for this PR.

**Which issue(s) this PR fixes**:

Related to #60957

**Special notes for your reviewer**:

<details><summary>Screenshots</summary>

<img width="1244" alt="Screenshot 2023-03-22 at 16 41 25" src="https://user-images.githubusercontent.com/745532/226963351-dc3d5290-a52d-4d90-ac3f-5549eae3b528.png">
<img width="1259" alt="Screenshot 2023-03-22 at 16 41 33" src="https://user-images.githubusercontent.com/745532/226963364-f6c7744c-eb1a-443e-a1e8-57114d21e5bc.png">
<img width="1244" alt="Screenshot 2023-03-22 at 16 41 44" src="https://user-images.githubusercontent.com/745532/226963376-33a647f8-f993-4c00-8f85-dac75ba1f288.png">
</details>

<details><summary>Videos</summary>

Adding new correlations:

https://user-images.githubusercontent.com/745532/227012407-6b18914d-d9c4-4e21-aa0b-b35579abf7b6.mov

Edit existing correlation:

https://user-images.githubusercontent.com/745532/227012449-556eb796-213f-4196-9206-42e967671878.mov

View read-only correlation:

https://user-images.githubusercontent.com/745532/227012471-ae1a4db8-eda5-4f0d-8850-d0ca65ceeba9.mov

<details>

<details><summary>How to test it</summary>

* Add a new correlation. Try not filling in all the fields to see validation messages.
* Edit an existing correlation. Check read-only (provisioning) and editable correlations.
* Use Back/Next buttons
</details>
